hooks/useL2Gas.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC, useState, useEffect, useMemo, createContext, useContext, ReactNode } from 'react';
+import BigNumber from 'bignumber.js';
+import { useRecoilValue } from 'recoil';
+import Connector from 'containers/Connector';
+import { networkState, isL2State, walletAddressState } from 'store/wallet';
+import { makeContract as makeL2WETHContract } from 'contracts/L2WETH';
+import { toBigNumber } from 'utils/formatters/number';
+
+type ContextTypes = {
+	balance: BigNumber;
+	showGetL2WETHPromptIfNone: () => boolean;
+	showGetWETHPrompt: boolean;
+	closeGetWETHPrompt: () => void;
+};
+const Context = createContext<ContextTypes | null>(null);
+
+type L2GasProviderProps = { children: ReactNode };
```

Comment:
hey @vbstreetz how about you create this one as a container? I know its essentially the same thing, but it makes using context a bit more "intuitive" (https://github.com/jamiebuilds/unstated-next)

Check the other containers, its minimal changes for you. Happy to walk you through the containers if you like.

Reply:
Awesome. I'll do the same on staking:loans 👍🏽